### ISSUE TITLE
[Q] Replicate Dense-SLAM Mapping

### ISSUE THREAD

**Issue by KyanainsGate** [YvanYin/Metric3D#49]
Hello, excellent work!!
By the way, may I please ask you about the Dense-SLAM experiment in the paper?

To my understanding, the weight for the "Droid" part is an off-the-shelf model that is trained on a TartanAir (released here: https://github.com/princeton-vl/DROID-SLAM), and the other hyperparameter is following the demo configuration of that repository. Is this correct?

In addition, I'm just curious about the input image: What the size of the image is fed into the Droid module? Or following one of the the cited papers, reshaped into a bit smaller size, like [192,640]?

Thank you very much for your support!

**Comment by JUGGHM** [comment]
> Hello, excellent work!! By the way, may I please ask you about the Dense-SLAM experiment in the paper?
> 
> To my understanding, the weight for the "Droid" part is an off-the-shelf model that is trained on a TartanAir (released here: https://github.com/princeton-vl/DROID-SLAM), and the other hyperparameter is following the demo configuration of that repository. Is this correct?
> 
> In addition, I'm just curious about the input image: What the size of the image is fed into the Droid module? Or following one of the the cited papers, reshaped into a bit smaller size, like [192,640]?
> 
> Thank you very much for your support!

Thanks for your question! I think @ckLibra should know this. Could you please help illustrate a little bit on this?

**Comment by KyanainsGate** [comment]
I appreciate all your kind and quick support!

If so, @ckLibra,  could you share with us the configuration (especially,  resolution) to get the number of Table 5 in the paper (e.g. `Droid+Ours` shows `t_rel=1.63`, etc.)?  
[192, 640] for it showed a completely worse result in my experiments so I believe that is too small or is harmed by a domain shift, but not sure what should be. 

Thank you!

**Comment by KyanainsGate** [comment]
Thank you for your time and great feedback, @ckLibra! 

I experimented with resolution [288, 960] just now. 
That says resolution gives no large difference in metrics, so we may have a 'better' configuration to work this on KITTI. 

Anyway, I'd love to wait for @JUGGHM's response for it. 
Thank you very much for all great support. 

P.S.
Let me share the experimental result: I'm 90% sure that no problem with the intrinsics alignment and t_rel calculation, so started to think like the above. 
![image](https://github.com/YvanYin/Metric3D/assets/29393239/9d7d41b5-bfb0-4098-a6cb-8f0f69c2cb9e)

| Result | resolution | t_rel | r_rel | `fx,fy,cx,cy` |
| ---- | ----  | ---- | ---- | ---- |
| Paper (**Droid**. w/ Metric3D) | ? | 21.7 | 0.23 | [553.7, 550.4, 471.3, 142.5] |
| Replicate A | [288,960] | 79.0 |  32.3 | [553.7, 550.4, 471.3, 142.5] |
| Replicate B | [192,640] | 77.1 | 31.4 |[369.1, 366.9, 314.2,  95.0] |

**Comment by JUGGHM** [comment]
> Thank you for your time and great feedback, @ckLibra!
> 
> I experimented with resolution [288, 960] just now. That says resolution gives no large difference in metrics, so we may have a 'better' configuration to work this on KITTI.
> 
> Anyway, I'd love to wait for @JUGGHM's response for it. Thank you very much for all great support.
> 
> P.S. Let me share the experimental result: I'm 90% sure that no problem with the intrinsics alignment and t_rel calculation, so started to think like the above. ![image](https://private-user-images.githubusercontent.com/29393239/297246335-9d7d41b5-bfb0-4098-a6cb-8f0f69c2cb9e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU0ODMzNjUsIm5iZiI6MTcwNTQ4MzA2NSwicGF0aCI6Ii8yOTM5MzIzOS8yOTcyNDYzMzUtOWQ3ZDQxYjUtYmZiMC00MDk4LWE2Y2ItOGYwZjY5YzJjYjllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE3VDA5MTc0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczYzAwY2FjOWI1ZDY1MDRmM2YxN2UwZjA4MDRlZTFiMTc0NWQ5YjM2Y2RhZmNmNmMwNDk4ZTg2MDkxZDNkN2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.saR9wFjH4TuCoXmbLjoXR9mQ0-oprUUZhnP5z_siCMk)
> 
> Result	resolution	t_rel	r_rel	`fx,fy,cx,cy`
> Paper (**Droid**. w/o Metric3D)	?	21.7	0.23	?
> Replicate A: Just RGB	[288,960]	79.0	32.3	[553.7, 550.4, 471.3, 142.5]
> Replicate B: Just RGB	[192,640]	77.1	31.4	[369.1, 366.9, 314.2, 95.0]

I checked the code of resize, it is:
`h1 = sqrt(384* 512 * h0 / w0)`
`w1 = sqrt(384* 512 * w0 / h0)`
where h0 and w0 are the original size.

So the size should be something close to （240,  824）

**Comment by ckLibra** [comment]
> Thank you for your time and great feedback, @ckLibra!
> 
> I experimented with resolution [288, 960] just now. That says resolution gives no large difference in metrics, so we may have a 'better' configuration to work this on KITTI.
> 
> Anyway, I'd love to wait for @JUGGHM's response for it. Thank you very much for all great support.
> 
> P.S. Let me share the experimental result: I'm 90% sure that no problem with the intrinsics alignment and t_rel calculation, so started to think like the above. ![image](https://private-user-images.githubusercontent.com/29393239/297246335-9d7d41b5-bfb0-4098-a6cb-8f0f69c2cb9e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU1NDg5NjQsIm5iZiI6MTcwNTU0ODY2NCwicGF0aCI6Ii8yOTM5MzIzOS8yOTcyNDYzMzUtOWQ3ZDQxYjUtYmZiMC00MDk4LWE2Y2ItOGYwZjY5YzJjYjllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTE4VDAzMzEwNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTYxNTg1ZGM0MTgxMmNkOTgwNTc5MGVjMzY3MzBiMTdjZGNjNWFhNDMwMGJmN2Y2Y2Y1ZDMzMWZkYWNlOWQ3ZGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AJc5E99rXDprsAtozJlaj2fBiZuXoAgxzn44GsSp92s)
> 
> Result	resolution	t_rel	r_rel	`fx,fy,cx,cy`
> Paper (**Droid**. w/o Metric3D)	?	21.7	0.23	?
> Replicate A: Just RGB	[288,960]	79.0	32.3	[553.7, 550.4, 471.3, 142.5]
> Replicate B: Just RGB	[192,640]	77.1	31.4	[369.1, 366.9, 314.2, 95.0]

I don't remember whether I registered the estimated trajectory to the gt before calculating the metric, but from your figure, it seems that this may be the reason for the poor r_rel.

**Comment by KyanainsGate** [comment]
Hello, sorry for the late response! 

> h1 = sqrt(384* 512 * h0 / w0)
> w1 = sqrt(384* 512 * w0 / h0)

Thank you @JUGGHM!
That seems similar to the eth3d demo in DROID official implementations. 

How about the metrics, such as scale alignment or each `rel` calculation?
Though I've just tried two eval scripts by the following repositories, both generate a very similar result and never got close to `(21.7, 0.23)` 
- https://github.com/Huangying-Zhan/kitti-odom-eval
- https://github.com/TRI-ML/KP3D/blob/master/kp3d/externals/cpp/evaluate_odometry.cpp

Thank you!

**Comment by JUGGHM** [comment]
> Hello, sorry for the late response!
> 
> > h1 = sqrt(384* 512 * h0 / w0)
> > w1 = sqrt(384* 512 * w0 / h0)
> 
> Thank you @JUGGHM! That seems similar to the eth3d demo in DROID official implementations.
> 
> How about the metrics, such as scale alignment or each `rel` calculation? Though I've just tried two eval scripts by the following repositories, both generate a very similar result and never got close to `(21.7, 0.23)`
> 
> * https://github.com/Huangying-Zhan/kitti-odom-eval
> * https://github.com/TRI-ML/KP3D/blob/master/kp3d/externals/cpp/evaluate_odometry.cpp
> 
> Thank you!

Any ideas on that @ckLibra ? How can I provide the information he might need according to our code space?

**Comment by JUGGHM** [comment]
> Hello, sorry for the late response!
> 
> > h1 = sqrt(384* 512 * h0 / w0)
> > w1 = sqrt(384* 512 * w0 / h0)
> 
> Thank you @JUGGHM! That seems similar to the eth3d demo in DROID official implementations.
> 
> How about the metrics, such as scale alignment or each `rel` calculation? Though I've just tried two eval scripts by the following repositories, both generate a very similar result and never got close to `(21.7, 0.23)`
> 
> * https://github.com/Huangying-Zhan/kitti-odom-eval
> * https://github.com/TRI-ML/KP3D/blob/master/kp3d/externals/cpp/evaluate_odometry.cpp
> 
> Thank you!

Currently I am not available to re-implement the droid-slam experiment myself. It was originally done by @ckLibra. After finishing Metric3D v2 I will closely examine this since many users propose issues on it.
